sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -422,14 +422,8 @@ static int deepHashCode(Object a, Schema.FieldType fieldType) {
         V value = e.getValue();
         V otherValue = b.get(key);
 
-        if (value == null) {
-          if (otherValue != null || !b.containsKey(key)) {
-            return false;
-          }
-        } else {
-          if (!deepEquals(value, otherValue, valueType)) {
-            return false;
-          }
+        if (!b.containsKey(key) || !deepEquals(value, otherValue, valueType)) {
```

Comment:
This is now more expensive because we now do two map lookups for all values while we used to only do two map lookups when the value was null.

Reply:
Right, I'm thinking about reverting this change and leaving tests only